src/encoder.rs

```diff
@@ -232,6 +232,11 @@ impl<W: Write> Encoder<W> {
 
     fn write_encoded_image_block(writer: &mut W, data_with_min_code_size: &[u8]) -> Result<(), EncodingError> {
         let (&min_code_size, data) = data_with_min_code_size.split_first().unwrap_or((&2, &[]));
+
+        // Check invariant in case invalid data was passed to `write_lzw_pre_encoded_frame`
```

Comment:
Ok, I looked into this some more. Because this only applies to `write_lzw_pre_encoded_frame`, if we're going to try to validate the LZW bitstream I think it makes sense for the check to happen there